lib/Dialect/FIRRTL/Transforms/LowerAnnotations.cpp

```diff
@@ -258,13 +258,7 @@ static LogicalResult applyDontTouch(const AnnoPathValue &target,
     return failure();
   }
 
-  // If the annotation is on a MemoryPortOp or if the annotation is on part of
-  // an aggregate, then keep the DontTouchAnnotation around.
-  if (isa<chirrtl::MemoryPortOp>(target.ref.getOp()) || target.fieldIdx)
-    return applyWithoutTarget<true>(target, anno, state);
-
-  target.ref.getInnerSym(state.getNamespace(target.ref.getModule()));
-  return success();
+  return applyWithoutTarget<true>(target, anno, state);
```

Comment:
This method `applyDontTouch` can be removed and replaced with a complete list of everything that it is allowed to apply to, i.e., `applyWithoutTarget<true, WireOp, NodeOp, MemOp, ...>`.